tensorflow_datasets/core/example_serializer_test.py

```diff
@@ -201,6 +201,23 @@ def test_add_ragged_fields_single_level_sequence(self):
     ])
     self.assertEqual(out[1], tensor_info)
 
+  def test_item_to_tf_feature_incorrect_shape(self):
+    # Test shape check in _item_to_tf_feature raises ValueError.
+    example_item = [1, 2, 3, 4, 5]
+    tensor_info = feature_lib.TensorInfo(shape=(4,), dtype=tf.int64)
+    with self.assertRaises(ValueError):
+      example_serializer._item_to_tf_feature(example_item, tensor_info)
+
+  def test_item_to_tf_feature_string_check(self):
+    # Test string check in _item_to_tf_feature raises ValueError.
+    example_item = [1, 2, 3, 4, 5]
+    tensor_info = feature_lib.TensorInfo(shape=(5,), dtype=tf.string)
+    with self.assertRaisesRegex(
+        ValueError,
+        'Unsupported value: (.*)\nCould not convert to bytes list.',
```

Comment:
Small note: This searches for a partial match. If trying to find a full match, include the start/end special chars (`^``$`). In this particular case it doesn't matter too much, but I just wanted to make sure you're aware.

Reply:
Got it. I will keep that in mind.